### PR TITLE
Overhaul recipes to replace iron-stick with iron-chest and concrete

### LINKED ISSUE #106: Warehouses require Iron Sticks which is not an unlocked intermediate
Space Age (or possibly 2.0 in general) changed Iron Sticks to not be unlocked from early, and requires a technology (Circuit Network, Concrete, Power Poles 1, Railway)

As warehousing may be unlocked before these, they're not usable without the Iron Stick Recipe.

Recommend adding the Iron Stick recipe to the Warehouse technology to allow them to be crafted without other technology requirements

### PR DESCRIPTION
**Pull request by dgw**
This resolves #106, though I did go a bit further than what was 'spitballed' there. Since we need to change the recipe anyway if we want to eliminate iron-stick, might as well take another look at the recipe balance at the same time.

Milestone 0.7.0 might become 1.0.0; it depends on how big this recipe rebalance ends up "feeling" if it gets bikeshedded.

----

In Factorio 2.0, iron-stick is an unlockable recipe, not automatically available at the start of the game. Adding iron-stick to every early technology that wants to use it in the resulting recipes feels hacky (though the base game does this; no less than 4 technologies unlock the iron-stick recipe now).

Crafting basic warehouse & storehouse has a relatively unchanged total iron cost, but iron-stick and some of the steel-plate in their recipes have been replaced with iron-chest.

Logistic variants are now intentionally more costly to craft, requiring multiples of the logistic chests they are to emulate (instead of only one), adding concrete, and increasing the steel-plate count.

Concrete also added to warehouse-logistics-research-1 prerequisites. Note that no migration is added to retroactively apply this change; players who have already unlocked this technology will keep it, but can't craft any more logistic buildings until they unlock Concrete.

_My lore-based reason for adding concrete to logistic buildings is that they probably have some automated item handling inside that needs the smoother floor, and can't deal with the roughness of stone brick. In reality of course, the reason is wanting to bump the complexity and cost a little more in exchange for the convenience._

### CODE REVIEW

File: prototypes/recipe.lua [hunk 1]
```diff
@@ -25,9 +25,9 @@ data:extend({
 		ingredients =
 		{
 			{ type = "item", name = "warehouse-basic", amount = 1 },
-			{ type = "item", name = "passive-provider-chest", amount = 1 },
-			{ type = "item", name = "steel-plate", amount = 10 },
-			{ type = "item", name = "iron-stick", amount = 15 },
+			{ type = "item", name = "passive-provider-chest", amount = 5 },
+			{ type = "item", name = "steel-plate", amount = 25 },
```
Comment (Anoyomouse) [inline prototypes/recipe.lua:29]: I had a chat with Nova (he's the one who initially planned out the mod with me)
How about a suggestion for:
- 1 of the logistic chests (for the port / logistics version)
- 25x advanced-circuit to fill out the requirements for the logistics stuff in the warehouse

This bolsters the copper requirement of the warehouses, which we both feel is a bit lacking at this stage, and would balance out the requirements

Also, requiring more steel doesn't make that much sense in lore, as your warehouse is already built, so you wouldn't need more steel

Comment (Anoyomouse) [reply prototypes/recipe.lua:29]: That being said, we're both happy with the requirement of concrete for (as you said) the smoother floor for the logistics system

**Comment by Anoyomouse** [comment]
Would also possibly bump this to a v1.0.0 with these changes, as it's pretty significant

**Comment by dgw** [comment]
> Would also possibly bump this to a v1.0.0 with these changes, as it's pretty significant

Wouldn't you know, I thought about that too 😸:

> Milestone 0.7.0 might become 1.0.0; it depends on how big this recipe rebalance ends up "feeling" if it gets bikeshedded.

We should go directly from 0.6.0 to 1.0.0 IMO. There are just a few things left to finish up, and I wouldn't hold up a new release for #108; another manual release or two will be fine.

Next thing I work on for Warehousing will be this recipe overhaul. I like the advanced circuit idea!

File: prototypes/recipe.lua (hunk 1 above)
Comment (dgw) [reply prototypes/recipe.lua]: I see that GitHub sees a conflict where local `git` can just adapt to the renamed files.

Would like to settle the recipes before I start rewriting any history, though.

And on that note, I have pushed my interpretation of how to apply advanced-circuit to logistic storehouses also (scaled down similarly to the concrete cost).

**Comment by Anoyomouse** [comment]
Happy, you just have to resolve the conflict :)

**Comment by Anoyomouse** [comment]
I went ahead and resolved the conflicts :)

**Comment by dgw** [comment]
> I went ahead and resolved the conflicts :)

Thanks, but merge commits inside feature branches are awful. Better to rebase, or manually merge on the CLI (like I'm about to). GitHub is being stupid about what's changed on `master`, but Git itself can handle it just fine.

**Pull request merged**